src/app/triggerFromForm.ts

```diff
@@ -9,14 +11,14 @@ interface TriggerFromFrom {
 }
 
 export class TriggerFromFormImpl implements TriggerFromFrom {
-    values: AppFormValues;
+    values: AppCallValues;
     context: AppContext;
-    trigger: any
+    trigger: ZDTrigger
 
-    constructor(call: AppCall) {
-        this.values = call.values as AppFormValues;
+    constructor(call: AppCallRequest) {
+        this.values = call.values as AppCallValues;
         this.context = call.context;
-        this.trigger = {};
+        this.trigger = {} as ZDTrigger;
```

Comment:
Does it make sense to use `?ZDTrigger` as the type for `trigger` instead?

Reply:
@hanzei, I'm not quite sure I understand your suggestion. I understand the ability to define a field as optional, using the `?` prefix, but I don't know how to use that in the context of defining an empty object for `ZDTrigger`

Reply:
After reading the class a second time I think it should be restructured. There is separate discussion about of the object oriented pattern is really needed, but I'm not going into that.

My suggestion is to the generation of the trigger into the constructor and letting `getTrigger` just returned the class field `trigger`.

Reply:
done.